<?xml version="1.0"?>
<robot name="arm">

    <link name="base_link"/>
    <joint name="shoulder_y_joint" type="revolute">
        <axis xyz="0 1 0"/>
        <limit effort="1000.0" lower="-6.2831" upper="6.2831" velocity="0.5" />
        <origin xyz="0.119 -0.195 -0.020" rpy="0 0 0"/>
        <parent link="base_link"/>
        <child link="shoulder_y_link"/>
    </joint>

    <link name="shoulder_y_link"/>
    <joint name="shoulder_z_joint" type="revolute">
        <axis xyz="0 0 1"/>
        <limit effort="1000.0" lower="-6.2831" upper="6.2831" velocity="0.5"/>
        <origin xyz="0 0 0" rpy="0 0 0"/>
        <parent link="shoulder_y_link"/>
        <child link="shoulder_z_link"/>
    </joint>
    <link name="shoulder_z_link"/>
    <joint name="shoulder_x_joint" type="revolute">
        <axis xyz="1 0 0"/>
        <limit effort="1000.0" lower="-6.2831" upper="6.2831" velocity="0.5" />
        <origin xyz="0 0 0" rpy="0 0 0"/>
        <parent link="shoulder_z_link"/>
        <child link="shoulder_x_link"/>
    </joint>
    <link name="shoulder_x_link"/>
    <joint name="elbow_z_joint" type="revolute">
        <axis xyz="0 0 1"/>
        <limit effort="1000.0" lower="-6.2831" upper="6.2831" velocity="0.5"/>
        <origin xyz="-0.26 0 0" rpy="0 0 0"/>
        <parent link="shoulder_x_link"/>
        <child link="elbow_z_link"/>
    </joint>
    <link name="elbow_z_link"/>
    <joint name="elbow_x_joint" type="revolute">
        <axis xyz="1 0 0"/>
        <limit effort="1000.0" lower="-6.2831" upper="6.2831" velocity="0.5"/>
        <origin xyz="0 0 0" rpy="0 0 0"/>
        <parent link="elbow_z_link"/>
        <child link="elbow_x_link"/>
    </joint>
    <link name="elbow_x_link"/>
    <joint name="wrist_y_joint" type="revolute">
        <axis xyz="0 1 0"/>
        <limit effort="1000.0" lower="-6.2831" upper="6.2831" velocity="0.5"/>
        <origin xyz="-0.26 0 0" rpy="0 0 0"/>
        <parent link="elbow_x_link"/>
        <child link="wrist_y_link"/>
    </joint>
    <link name="wrist_y_link"/>
    <joint name="end_fixed" type="fixed">
        <parent link="wrist_y_link"/>
        <child link="end_link"/>
        <origin xyz="-0.08 0.0 0" rpy="0 0 0"/>
    </joint>
    <link name="end_link"/>

    <!-- <joint name="imu_1_fixed" type="fixed">
        <parent link="shoulder_z_link"/>
        <child link="imu_1_link"/>
        <origin xyz="0 0.04 0.2" rpy="0 1.57079 1.57079"/>
    </joint>
    <link name="imu_1_link"/>
    <joint name="imu_2_fixed" type="fixed">
        <parent link="elbow_z_link"/>
        <child link="imu_2_link"/>
        <origin xyz="0 0.02 0.25" rpy="0 1.57079 1.57079"/>
    </joint>
    <link name="imu_2_link"/> -->



</robot>